From 93d95ed2eaedcca110c214e1fe3f8896b1f6f853 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Tue, 17 Dec 2019 20:24:27 +0100
Subject: [PATCH] xattr: adjust for attr 2.4.48 release

Latest versions of attr have removed the xattr.h header,
with the rationale that libc is providing the same wrappers.

attr/attributes.h is providing the ENOATTR definition.

Upstream-Status: Pending
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 ports/linux/subports         | 5 +++--
 ports/linux/xattr/portdefs.h | 3 ++-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/ports/linux/subports b/ports/linux/subports
index 2c43ac9..740ec83 100755
--- a/ports/linux/subports
+++ b/ports/linux/subports
@@ -29,11 +29,12 @@ fi
 if	$port_xattr; then
 	cat > dummy.c <<EOF
 #include <sys/types.h>
-#include <attr/xattr.h>
+#include <sys/xattr.h>
+#include <attr/attributes.h>
 int i;
 EOF
 	if ! ${CC} -c -o dummy.o dummy.c >/dev/null 2>&1; then
-		echo >&2 "Warning: Can't compile trivial program using <attr/xattr.h>".
+		echo >&2 "Warning: Can't compile trivial program using <attr/attributes.h>".
 		echo >&2 "         xattr support will require that header."
 	fi
 	echo "linux/xattr"
diff --git a/ports/linux/xattr/portdefs.h b/ports/linux/xattr/portdefs.h
index 56cd3ca..068d39a 100644
--- a/ports/linux/xattr/portdefs.h
+++ b/ports/linux/xattr/portdefs.h
@@ -2,5 +2,6 @@
  * SPDX-License-Identifier: LGPL-2.1-only
  *
  */
-#include <attr/xattr.h>
+#include <sys/xattr.h>
+#include <attr/attributes.h>
 #include <stdint.h>
